src/meteors/shap/explanation.py

```diff
@@ -0,0 +1,136 @@
+    elif not isinstance(data, np.ndarray):
+        raise TypeError(f"Expected np.ndarray, pd.DataFrame, or torch.Tensor as data, but got {type(data)}")
+    return data
+
+
+def process_and_validate_explanations(explanations: shap.Explanation, info: ValidationInfo) -> shap.Explanation:
+    if not isinstance(explanations, shap.Explanation):
+        raise TypeError(f"Expected shap.Explanation as explanations, but got {type(explanations)}")
+
+    if "data" not in info.data:
+        raise RuntimeError("ValidationInfo object is broken - pydantic internal error")
+
+    data_shape = info.data["data"].shape
+    # check if the shape of the explanations is correct
+    if explanations.shape != data_shape:
+        # only the last dimension differs
+        if explanations.shape[:-1] == data_shape:
+            logger.debug("Detected explanation for multiple targets based on the explanation shape validation.")
+        else:
+            raise ShapeMismatchError(
+                f"Shape of the explanations does not match the shape of the input data. Expected {data_shape}, but got {explanations.shape}"
+            )
+
+    return explanations
+
+
+###################################################################
+########################### EXPLANATION ###########################
+###################################################################
+
+
+class SHAPExplanation(BaseModel):
+    """Represents an object that contains SHAP explanations for a model.
+
+    Args:
+        data (np.ndarray): a numpy array containing the input data.
+        explanations (np.ndarray): a numpy array containing the SHAP explanations.
+        If the model outputs a single value, the shape of the array should be equal to the shape of the input data.
+        In case the model outputs multiple values, the last dimension of the array should be equal to the number of outputs and the rest of the dimensions should be equal to the input data.
+        explanation_method (str): the method used to generate the explanation.
+        feature_names (list[str]): a list of feature names.
+        aggregations (list[list[str]] | list[str]): a list of feature aggregations.
+    """
+
+    data: Annotated[
+        np.ndarray | torch.Tensor | pd.DataFrame,
+        PlainValidator(ensure_data_type),
+        Field(
+            description="A numpy array containing the input data.",
+        ),
+    ]
+    explanations: Annotated[
+        shap.Explanation,
+        PlainValidator(process_and_validate_explanations),
+        Field(
+            description="A numpy array containing the SHAP explanations.",
+        ),
+    ]
+    explanation_method: Annotated[
+        str | None,
+        PlainValidator(ensure_explainer_type),
+        Field(
+            description="The method used to generate the explanation.",
+        ),
+    ] = None
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    @property
+    def target_dims(self) -> int:
```

Comment:
Please write more informative description, don't understand what does it return.

Reply:
added additional description, renamed this field to `num_target_outputs`, [b43bce7](https://github.com/xai4space/meteors/pull/102/commits/b43bce796561bfb2e0eac933a97dbe29044bf396)